msprime/simulations.py

```diff
@@ -1471,6 +1489,198 @@ def print_history(self, output=sys.stdout):
             self._print_populations(epoch, output)
             print(file=output)
 
+    def population_size_trajectory(self, steps):
+        """
+        This function returns an array of per-population effective population sizes,
+        as defined by the demographic model. These are just the `population_size`
+        parameters of the model, modified by any population growth rates.
+        The sizes are computed at a regular grid of time steps, between time 0 and `end`
+        units of time ago.
+
+        :param int end: Time (ago) at which to stop trajectory inference.
+
+        :param int num_steps: determines at how many time points the
+            coalescence rate is computed.
+
+        :return: Returns a tuple where the first value
+            is an array giving the times at which population sizes have been computed,
+            the second is a num_steps-by-num_pops ndarray containing the effective
+            population sizes for each subpopulation individually,
+            at the corresponding step
+
+        :rtype: :class:'tuple'
+        """
+        num_pops = self.num_populations
+        N_t = np.zeros([len(steps), num_pops])
+        for j, t in enumerate(steps):
+            N, _ = self.pop_size_and_migration_at_t(t)
+            N_t[j] = N
+        return N_t
+
+    def coalescence_rate_trajectory(self, steps, num_samples, min_pop_size=1,
+                                    double_step_validatation=True):
+        """
+        This function will calculate the mean coalescence rates between the
+        lineages of the sample configuration provided in `num_samples`, at
+        each generation time listed by `steps`, in this demographic model.
+        The coalescence rate at time t in the past is the average rate
+        of coalescence of as-yet-uncoalesed lineages, computed as follows:
+        let p(t) be the probability that the lineages of a randomly
+        chosen pair of samples has not yet coalesced by time t,
+        let p(z,t) be the probability that the lineages of a randomly
+        chosen pair of samples has not yet coalesced by time t *and* are
+        both in population z, and let N(z,t) be the diploid effective
+        population size of population z at time t. Then the mean coalescence
+        rate at time t is r(t) = (sum_z p(z,t) / (2 * N(z,t) ) / p(t).
+        The computation is done approximating population size trajetories
+        with piecewise constant trajectories, in steps of size dt = end / num_steps.
+        For this to be accurate, dt must be small enough so that (a) short epochs
+        (e.g., bottlenecks) are not missed, and (b) populations do not
+        change in size too much over that time, if they are growing or shrinking.
+        If in doubt, check that results are simliar after doubling `num_steps`.
```

Comment:
Instead of "If in doubt..", 
"This function optionally provides a simple check of this approximation by recomputing the coalescence rates on a grid of steps twice as fine and throwing a warning if the resulting values do not match to a relative tolerance of 0.001.